addons/web/static/src/js/views/control_panel/control_panel_model.js

```diff
@@ -1018,6 +1021,49 @@ var ControlPanelModel = mvc.Model.extend({
         }
         return context;
     },
+    /**
+     * Insert a combination of type [filterId, optionId] in activeFilterIds for
+     * a corresponding filter of type 'groupBy' with options.
+     * The array activeFilterIds is by construction of the form
+     *
+     * A1 'concat' A2 'concat' A3
+     *
+     * where filterId does not occur in the arrays A1 and A3 (possibly empty) and
+     * A2 = [[filterId, optionId_1], ..., [filterId, optionId_n]] (possibly empty).
+     *
+     * The idea of the function below is to insert [filterId, optionId] at the end
+     * of activeFilterIds if A2 is empty and inside A2 else while keeping A2 ordered
+     * for the natural order on option ids: 'year' < 'quarter' < ... < 'day'.
+     *
+     * @param {Array[]} activeFilterIds
+     * @param {Array} combinationId
+     */
+    _insert(activeFilterIds, combinationId) {
+        const filterId = combinationId[0];
+        let firstIndex = -1;
+        let lastIndex = -1;
+        activeFilterIds.forEach((cId, i) => {
+            if (cId[0] === filterId) {
+                if (firstIndex === -1) {
+                    firstIndex = i;
+                }
+                lastIndex = i;
+            }
+        });
+        if (firstIndex === -1) {
```

Comment:
if (firstIndex === -1) { // filterId not in activeFilterIds yet

Reply:
ok